docs/develop/authorize-to-microsoft-graph.md

```diff
@@ -59,3 +59,16 @@ For examples of detailed walkthroughs and scenarios, see:
 * [Create a Node.js Office Add-in that uses single sign-on](create-sso-office-add-ins-nodejs.md)
 * [Create an ASP.NET Office Add-in that uses single sign-on](create-sso-office-add-ins-aspnet.md)
 * [Scenario: Implement single sign-on to your service in an Outlook add-in](../outlook/implement-sso-in-outlook-add-in.md)
+
+## Distributing SSO-enabled add-ins in Microsoft AppSource
+
+When an Office 365 admin acquires an add-in from [AppSource](https://appsource.microsoft.com), the admin can redistribute it by [centralized deployment](../publish/centralized-deployment.md) and grant admin consent to the add-in to access Microsoft Graph scopes. It is also possible, however, for an end user to acquire the add-in directly from AppSource, in which case the user must grant consent to the add-in. This can create a potential performance problem for which we've provided a solution.
+
+If your code passes the `allowConsentPrompt` option in the call of `getAccessToken`, like this, `OfficeRuntime.auth.getAccessToken( { allowConsentPrompt: true } )`; then Office can prompt the user for consent if Azure AD reports to Office that consent has not yet been granted to the add-in. However, for security reasons, Office can only prompt the user to consent to the Azure AD `profile` scope. *Office cannot prompt for consent to any Microsoft Graph scopes*, not even `User.Read`. This means that if the user grants consent on the prompt, Office will return a bootstrap token. But the attempt to exchange the bootstrap token for an access token to Microsoft Graph will fail with error AADSTS65001, which means consent (to Microsoft Graph scopes) has not been granted.
+
+Your code can, and should, handle this error by falling back to an alternate system of authentication, which will prompt the user for consent to Microsoft Graph scopes. (For code examples, see [Create a Node.js Office Add-in that uses single sign-on](create-sso-office-add-ins-nodejs.md) and [Create an ASP.NET Office Add-in that uses single sign-on](create-sso-office-add-ins-aspnet.md) and the samples they link to.) But the entire process requires multiple round trips to Azure AD. You can avoid this performance penalty by passing the `forMSGraphAccess` option in the call of `getAccessToken`; for example, `OfficeRuntime.auth.getAccessToken( { forMSGraphAccess: true } )`.  This signals to Office that your add-in needs Microsoft Graph scopes. Office will ask Azure AD to verify that consent to Microsoft Graph scopes has already been granted to the add-in. If it has, the bootstrap token will be returned. If it hasn't, then the call of `getAccessToken` will return error 13012. Your code can handle this error by falling back to an alternate system of authentication immediately, without making a doomed attempt to exchange tokens with Azure AD.
```

Comment:
```suggestion
Your code can, and should, handle this error by falling back to an alternate system of authentication, which will prompt the user for consent to Microsoft Graph scopes. For code examples, see [Create a Node.js Office Add-in that uses single sign-on](create-sso-office-add-ins-nodejs.md) and [Create an ASP.NET Office Add-in that uses single sign-on](create-sso-office-add-ins-aspnet.md) and the samples they link to. The entire process requires multiple round trips to Azure AD. You can avoid this performance penalty by passing the `forMSGraphAccess` option in the call of `getAccessToken`; for example, `OfficeRuntime.auth.getAccessToken( { forMSGraphAccess: true } )`.  This signals to Office that your add-in needs Microsoft Graph scopes. Office will ask Azure AD to verify that the add-in has granted consent to Microsoft Graph scopes. If it has, the bootstrap token will be returned. If it hasn't, then the call of `getAccessToken` will return error 13012. Your code can handle this error by falling back to an alternate system of authentication immediately, without making a doomed attempt to exchange tokens with Azure AD.
```

Reply:
This suggestion changes the meaning. The add-in does not grant consent, rather a user or admin grants consent to it. Also, I prefer having the links to the samples parenthetical.

Reply:
Then leave as is for sure! MS Style states _someplace_ that "if it's important enough to include, then include it and leave out the parentheses." [sic] I try to avoid parentheses and use the built in alerts or See Also section. Again, my comments on this article are SUPER MINOR so totally fine to ignore, if you wish. No show-stoppers!